haskell/test/Lorentz/Contracts/Test/Management.hs

```diff
@@ -0,0 +1,504 @@
+      let
+        originationParams =
+            addAccount (wallet1, (commonOperators, 10))
+          $ addAccount (wallet2, ([], 0))
+          $ defaultOriginationParams
+              { opPermissionsDescriptor = permissionDescriptorOperatorTransferAllowed }
+      withOriginated originate originationParams $ \stablecoinContract -> do
+        lCallEP stablecoinContract (Call @"Pause") ()
+        let
+          transfers =
+            [ (#from_ .! wallet1, (#to_ .! wallet2, (#token_id .! 0, #amount .! 10)))
+            ]
+
+        withSender commonOperator $ lCallEP stablecoinContract (Call @"Transfer") transfers
+
+        validate . Left $ lExpectAnyMichelsonFailed stablecoinContract
+
+    it "can successfully transfer tokens after contract unpause" $ integrationalTestExpectation $ do
+      let
+        originationParams =
+            addAccount (wallet1, (commonOperators, 10))
+          $ addAccount (wallet2, (commonOperators, 0))
+          $ addAccount (wallet3, (commonOperators, 0))
+          $ defaultOriginationParams
+              { opPermissionsDescriptor = permissionDescriptorOperatorTransferAllowed
+              , opPaused = True
+              }
+
+      withOriginated originate originationParams $ \stablecoinContract -> do
+        lCallEP stablecoinContract (Call @"Unpause") ()
+        let
+          transfers =
+            [ (#from_ .! wallet1, (#to_ .! wallet2, (#token_id .! 0, #amount .! 10)))
+            , (#from_ .! wallet2, (#to_ .! wallet3, (#token_id .! 0, #amount .! 10)))
+            ]
+
+        withSender commonOperator $ lCallEP stablecoinContract (Call @"Transfer") transfers
+
+        consumer <- lOriginateEmpty @[BalanceResponseItem] contractConsumer "consumer"
+        let
+          balanceRequestItems =
+            [ (#owner .! wallet1, #token_id .! 0)
+            , (#owner .! wallet2, #token_id .! 0)
+            , (#owner .! wallet3, #token_id .! 0)
+            ]
+          balanceRequest = mkView (#requests .! balanceRequestItems) consumer
+          balanceExpected =
+            [ (#request .! (#owner .! wallet1, #token_id .! 0), #balance .! 0)
+            , (#request .! (#owner .! wallet2, #token_id .! 0), #balance .! 0)
+            , (#request .! (#owner .! wallet3, #token_id .! 0), #balance .! 10)
+            ]
+
+        lCallEP stablecoinContract (Call @"Balance_of") balanceRequest
+
+        validate . Right $
+          lExpectViewConsumerStorage consumer [balanceExpected]
+
+
+
+  describe "Configure minter" $ do
+    it "configures minter properly" $ integrationalTestExpectation $ do
+      withOriginated originate defaultOriginationParams $ \stablecoinContract -> do
+        let
+          configureMinterParam1 =
+            ( #minter .! wallet2
+            , ( #current_minting_allowance .! Nothing
+              , #new_minting_allowance .! 20
+              ))
+
+        withSender admin $ lCallEP stablecoinContract (Call @"Configure_minter") configureMinterParam1
```

Comment:
Why is admin calling this entrypoint. Isn't this supposed to be called by master minter?

Reply:
We have no `admin` role in stablecoin contract. Here admin is an actual superuser that has all roles assigned to him. Regarding to permissions validation there's already present test (`fails if sender does not have master minter permissions`) for it.

Reply:
> Here admin is an actual superuser that has all roles assigned to him. 

Yes, I was referring to this. Basically, I think we should have different addresses for owner, master minter and pause roles. Or else this won't catch if the owner actually allows the owner (I mean the admin here), to call the entrypoints that are only callable by other two roles, and vice versa.